pkg/tcpip/tcpip.go

```diff
@@ -179,6 +179,36 @@ func (m AddressMask) Prefix() int {
 	return p
 }
 
+// Size returns the number of leading ones and total bits in the mask. If
+// the mask is not in the canonical form--ones followed by zeros--then Size
+// returns 0, 0.
+func (m AddressMask) Size() (ones, size int) {
+	// A subnet mask in canonical form consists of zero or more 1's followed by
+	// 0's up to a total length of 8 * len(m) bits. Refer to golang's net.IPMask.
+	var i int
+	for i = 0; i < len(m); i++ {
+		if m[i] != 0xff {
+			break
+		}
+	}
+	ones = i * 8
+	if i < len(m) {
+		leading1s := bits.LeadingZeros8(^m[i])
+		if leading1s != bits.OnesCount8(m[i]) {
+			return 0, 0
+		}
+		ones += leading1s
+		i++
+	}
+	for ; i < len(m); i++ {
```

Comment:
Add a comment here indicating that we are asserting that all trailing bytes have zeroes for the subnet to be in canonical form.